tchannel/tornado/dispatch.py

```diff
@@ -233,17 +234,36 @@ def _on_post_response(future):
             e.id = request.id
             connection.send_error(e)
         except Exception as e:
-            error = UnexpectedError(
-                description="Unexpected Error: '%s'" % e,
-                id=request.id,
-                tracing=request.tracing,
-            )
-            response.set_exception(error)
-            connection.request_message_factory.remove_buffer(response.id)
+            try:
+                exc_type, exc_obj, exc_tb = sys.exc_info()
```

Comment:
Nit: Generally, the assignment should take place outside the try block if you're using the value in the finally block.

    x = foo
    try:
        # ...
    finally:
        do_stuff_with(x)

That's because if it's inside the try block, and the block fails before assigning, the variable will be undefined inside the finally block. This doesn't apply as strongly here because `sys.exc_info()` will never fail, but it still makes sense to assign outside the block.